finetrainers/hunyuan_video/hunyuan_video_lora.py

```diff
@@ -0,0 +1,319 @@
+    cache_dir: Optional[str] = None,
+    enable_slicing: bool = False,
+    enable_tiling: bool = False,
+    enable_model_cpu_offload: bool = False,
+) -> HunyuanVideoPipeline:
+    component_name_pairs = [
+        ("tokenizer", tokenizer),
+        ("text_encoder", text_encoder),
+        ("tokenizer_2", tokenizer_2),
+        ("text_encoder_2", text_encoder_2),
+        ("transformer", transformer),
+        ("vae", vae),
+        ("scheduler", scheduler),
+    ]
+    components = {}
+    for name, component in component_name_pairs:
+        if component is not None:
+            components[name] = component
+
+    pipe = HunyuanVideoPipeline.from_pretrained(model_id, **components, revision=revision, cache_dir=cache_dir)
+    pipe.text_encoder = pipe.text_encoder.to(dtype=text_encoder_dtype)
+    pipe.text_encoder_2 = pipe.text_encoder_2.to(dtype=text_encoder_2_dtype)
+    pipe.transformer = pipe.transformer.to(dtype=transformer_dtype)
+    pipe.vae = pipe.vae.to(dtype=vae_dtype)
+
+    if enable_slicing:
+        pipe.vae.enable_slicing()
+    if enable_tiling:
+        pipe.vae.enable_tiling()
+
+    if enable_model_cpu_offload:
+        pipe.enable_model_cpu_offload(device=device)
+    else:
+        pipe.to(device=device)
+
+    return pipe
+
+
+def prepare_conditions(
+    tokenizer: LlamaTokenizer,
+    text_encoder: LlamaModel,
+    tokenizer_2: CLIPTokenizer,
+    text_encoder_2: CLIPTextModel,
+    prompt: Union[str, List[str]],
+    guidance: float = 1.0,
+    device: Optional[torch.device] = None,
+    dtype: Optional[torch.dtype] = None,
+    max_sequence_length: int = 128,
+    # TODO(aryan): make configurable
+    prompt_template: Dict[str, Any] = {
+        "template": (
+            "<|start_header_id|>system<|end_header_id|>\n\nDescribe the video by detailing the following aspects: "
+            "1. The main content and theme of the video."
+            "2. The color, shape, size, texture, quantity, text, and spatial relationships of the objects."
+            "3. Actions, events, behaviors temporal relationships, physical movement changes of the objects."
+            "4. background environment, light, style and atmosphere."
+            "5. camera angles, movements, and transitions used in the video:<|eot_id|>"
+            "<|start_header_id|>user<|end_header_id|>\n\n{}<|eot_id|>"
+        ),
+        "crop_start": 95,
+    },
+) -> torch.Tensor:
+    device = device or text_encoder.device
+    dtype = dtype or text_encoder.dtype
+
+    if isinstance(prompt, str):
+        prompt = [prompt]
+
+    conditions = {}
+    conditions.update(
+        _get_llama_prompt_embeds(tokenizer, text_encoder, prompt, prompt_template, device, dtype, max_sequence_length)
+    )
+    conditions.update(_get_clip_prompt_embeds(tokenizer_2, text_encoder_2, prompt, device, dtype))
+
+    guidance = torch.tensor([guidance], device=device, dtype=dtype) * 1000.0
+    conditions["guidance"] = guidance
+
+    return conditions
```

Comment:
Wonder if it's possible to leverage the `encode_prompt()` from the pipeline itself. TODO

Reply:
I think it's better to keep custom implementations here per model because it is clean to understand and debug without jumping to diffusers codebase. Also, our pipelines contain some additional things and checks at times - let's revisit this idea maybe later